File: Stage23/Java/Problem1541.java
package Stage23.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Problem1541 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();

        String[] minus = s.split("-");

        int result = 0;

        String[] first = minus[0].split("\\+");
        for (String num : first) {
            result += Integer.parseInt(num);
        }

        for (int i = 1; i < minus.length; i++) {
            String[] plus = minus[i].split("\\+");
            int temp = 0;
            for (String num : plus) {
                temp += Integer.parseInt(num);
            }
            result -= temp;
        }

        System.out.println(result);
        br.close();
    }
}

File: Stage23/C/Problem1541.c
#include <stdio.h>

int main(void) {
    char s[100];
    scanf("%s", s);

    int result = 0, num = 0, minus = 0;

    for (int i = 0; s[i]; i++) {
        if (s[i] >= '0' && s[i] <= '9') {
            num = num * 10 + (s[i] - '0');
        } else {
            if (minus)
                result -= num;
            else
                result += num;
            
            if (s[i] == '-')
                minus = 1;
            
            num = 0;
        }
    }

    if (minus)
        result -= num;
    else
        result += num;
    
    printf("%d\n", result);
    return 0;
}